src/external/dftk_to_wannier90.jl

```diff
@@ -0,0 +1,415 @@
+    k = basis.kpoints[ik]; kpb = basis.kpoints[ikpb] #renaming for clarity
+    map_fourier_modes = [ (iGk,DFTK.index_G_vectors(basis, kpb, Gk+G_shift))
+                          for (iGk,Gk) in enumerate(G_vectors(k))
+                          if !isnothing(DFTK.index_G_vectors(basis,kpb,Gk+G_shift)) ]
+
+    # Compute overlaps
+    for n in 1:n_bands
+        for m in 1:n_bands
+            # Select the coefficient in right order
+            Gk_coeffs = @view ψ[ik][[i[1] for i in map_fourier_modes],m] 
+            Gkpb_coeffs = @view ψ[ikpb][[i[2] for i in map_fourier_modes],n]
+            Mkb[m,n] = dot(Gk_coeffs,Gkpb_coeffs)
+        end
+    end    
+
+    Mkb
+end
+
+
+"""
+Iteratively use the preceeding function on each k and kpb to generate the whole .mmn file.
+"""
+function generate_mmn_file(prefix::String, basis::PlaneWaveBasis, ψ,
+                           nn_kpts, nn_num::Integer)
+    # general parameters
+    n_bands = size(ψ[1],2)
+    k_size = length(ψ)
+    
+    progress = Progress(size(nn_kpts,1), desc="Computing Mmn overlaps : ")
+    
+    # Small function for the sake of clarity
+    read_nn_kpts(n) = nn_kpts[n][1],nn_kpts[n][2],nn_kpts[n][3:end]
+
+    # Write file
+    open("$prefix.mmn", "w") do f
+        write(f,"Generated by DFTK at $(now())"*"\n")
+        write(f,"$n_bands  $k_size  $nn_num"*"\n")
+
+        # Loop over all (k_points, nearest_neighbour, shif_vector)
+        for i_nnkp in 1:size(nn_kpts,1)
+            # Write the label of the matrix
+            k,kpb,shift = read_nn_kpts(i_nnkp)
+            write(f,@sprintf("%i  %i  %i  %i  %i \n",
+                             k,kpb,shift[1],shift[2],shift[3]) )
+            # The matrix itself
+            Mkb = overlap_Mmn_k_kpb(basis, ψ, k, kpb, n_bands,; G_shift = shift)
+            for ovlp in Mkb
+                write(f, @sprintf("%22.18f %22.18f \n",real(ovlp),imag(ovlp)) )
+            end
+            next!(progress)
+        end
+    end
+    
+end
+
+
+@doc raw"""
+Compute the matrix ``[A_k]_{m,n} = \langle ψ_m^k | g^{\text{per}}_n \rangle`` 
+
+``g^{per}_n`` are periodized gaussians whose respective centers are given as an
+ (n_bands,1) array [ [center 1], ... ].
+Centers are to be given in lattice coordinates.
+
+The dot product is computed in the Fourier space. 
+
+Given an orbital ``g_n``, the periodized orbital is defined by :
+ ``g^{per}_n=  \sum\limits_{R in lattice} g_n( ⋅ - R)``. 
+``g^{per}_n`` is not explicitly computed. Its Fourier coefficient at any G 
+is given by the value of the Fourier transform of ``g_n`` in G.
+"""
+function A_k_matrix_gaussian_guesses(basis::PlaneWaveBasis, ψ, k::Integer,
+                                     n_bands::Integer, n_wann::Integer;
+                                     centers = [], coords = "", projs = [])
+
+    # associate a center with the fourier transform of the corresponding gaussian
+    guess_fourier(center) = xi ->  exp(-im*dot(xi,center) - dot(xi,xi)/4) 
```

Comment:
we typically call the Fourier variable `q`

Reply:
also this kind of function returning function is usually a big performance gotcha, and can be hard to read. Might as well do `guess_fourier(center, q) = ...`